### ISSUE TITLE
Update SQLite bundled with Docker container

### ISSUE THREAD

**Issue by simonw** [simonw/datasette#695]
It's 3.26.0 at the moment:
https://github.com/simonw/datasette/blob/af9cd4ca64652fae262e6f7b5d201f6e0adc989b/Dockerfile#L9-L11
Most recent release is 3.31.1: https://www.sqlite.org/releaselog/3_31_1.html

**Comment by simonw** [comment]
From https://www.sqlite.org/download.html the download URL is https://www.sqlite.org/2020/sqlite-autoconf-3310100.tar.gz

**Comment by simonw** [comment]
On my laptop I can build and test like so:

    cd datasette
    docker build -t datasette-sqlite .
    docker run -p 8001:8001 -v `pwd`:/mnt \
        datasette-sqlite \
        datasette -p 8001 -h 0.0.0.0 /mnt/fixtures.db

**Comment by simonw** [comment]
Need to test with SpatiaLite here too.
```
docker run -p 8001:8001 -v `pwd`:/mnt \
    datasette-sqlite \
    datasette -p 8001 -h 0.0.0.0 /mnt/spatial.db \
    --load-extension=/usr/local/lib/mod_spatialite.so
```

**Comment by simonw** [comment]
Here's how to run the unit test suite inside the freshly built Docker container (built with a new SQLite):
```
docker run -it -v `pwd`:/mnt datasette-sqlite /bin/bash
root@0e1928cfdf79:/# cd /mnt
root@0e1928cfdf79:/mnt# pip install -e .[test]
root@0e1928cfdf79:/mnt# pytest
```
I ran that against SQLite 3.31.1 and got one failure!

It was for `test_searchable[/fixtures/searchable.json?_search=te*+AND+do*&_searchmode=raw-expected_rows3]`
```
    def test_searchable(app_client, path, expected_rows):
        response = app_client.get(path)
>       assert expected_rows == response.json["rows"]
E       AssertionError: assert [[1, 'barry c...sel', 'puma']] == []
E         Left contains 2 more items, first extra item: [1, 'barry cat', 'terry dog', 'panther']
E         Full diff:
E         + []
E         - [[1, 'barry cat', 'terry dog', 'panther'],
E         -  [2, 'terry dog', 'sara weasel', 'puma']]
```

**Comment by simonw** [comment]
Well that's odd. If I run the tests against SQLite 3.26.0 inside the container I get the same failure there as well:

```
docker run -it -v `pwd`:/mnt datasetteproject/datasette:0.37.1 /bin/bash
root@710eed6278b2:/# cd /mnt
root@710eed6278b2:/mnt# pip install -e .[test] && pytest
```

**Comment by simonw** [comment]
SpatiaLite testing works fine. I'm going to ship this even though there's a single failing test, since that's not a regression from the previous Docker release. I'm going to file a separate issue about that.

**Comment by simonw** [comment]
Shipped in [Datasette 0.38](https://datasette.readthedocs.io/en/latest/changelog.html#v0-38).
